nanoFramework.Hardware.Esp32.Rmt/RmtChannelSettings.cs

```diff
@@ -0,0 +1,142 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// Base class for shared RMT Channel settings.
+    /// </summary>
+    public abstract class RmtChannelSettings
+    {
+        private int _channel;
+        private int _pinNumber;
+        private byte _clockDivider;
+        private byte _numberOfMemoryBlocks;
+        private int _bufferSize;
+        private bool _signalInverterEnabled;
+
+        /// <summary>
+        /// Gets or sets the channel number.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public int Channel
+        {
+            get => _channel;
+            set => _channel = value;
```

Comment:
I guess values are from 0 to 7, correct? If yes, then worth testing this as well? Maybe it's done on the native side

Reply:
Correct! This has been updated.